packages/admin-ui/ui/src/domain/orders/details/create-fulfillment/item-table.tsx

```diff
@@ -118,12 +119,32 @@ const FulfillmentLine = ({
     })
   }, [validQuantity, setErrors, item.id])
 
+  React.useEffect(() => {
+    if (!availableQuantity) {
+      handleQuantityUpdate(0, item.id)
+    } else {
+      handleQuantityUpdate(
+        Math.min(getFulfillableQuantity(item), availableQuantity),
+        item.id
+      )
+    }
+    // Note: we can't add handleQuantityUpdate to the dependency array as it will cause an infinite loop
```

Comment:
What if we useCallback-ed it? Ignoring exhaustive-deps is usually a bad sign ime

Reply:
Had a quick huddle around this, resolved that it's weirdly circular and not worth chasing a clean fix at the moment.